src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,28 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.CantAddUserException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws CantAddUserException {
+        if (storageDao.get(user.getLogin()) == null
+                && user.getLogin().length() >= 6
+                && user.getPassword().length() >= 6
+                && user.getAge() >= 18) {
```

Comment:
You dont need this condition, if you put this `if` block after validation `if- else if` blocks